### DISCUSSION TITLE
Get SSID Android

### DISCUSSION THREAD

**Discussion by renanpaiva64** [dotnet/maui#13707] [Q&A] [answered]
Hello everyone.

I need to get the Android SSID. But It's always returning unknown ssid. After some research at internet, I've found that it should be a permission issue, so I've added the required permissions at manifest and main without success.

Following the code I'm using.
               
               `var cm = (ConnectivityManager)Android.App.Application.Context.GetSystemService(Android.Content.Context
                                .ConnectivityService)!;
                var capabilities = cm.GetNetworkCapabilities(cm.ActiveNetwork);
                var info = (WifiInfo)capabilities.TransportInfo;
                return info?.SSID;`

Top of my main:

 [assembly: UsesPermission(Name = "android.permission.ACCESS_WIFI_STATE")]
 [assembly: UsesPermission(Name = "android.permission.ACCESS_FINE_LOCATION")]
 [assembly: UsesPermission(Name = "android.permission.ACCESS_NETWORK_STATE")]
 [assembly: UsesFeature("android.hardware.location", Required = false)]
 [assembly: UsesFeature("android.hardware.location.gps", Required = false)]
 [assembly: UsesFeature("android.hardware.location.network", Required = false)]

My manifest:

```
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
<uses-permission android:name="android.permission.INTERNET" />
<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
<uses-feature android:name="android.hardware.location" android:required="false" />
<uses-feature android:name="android.hardware.location.gps" android:required="false" />
<uses-feature android:name="android.hardware.location.network" android:required="false"/>
```

And finally the result:

![image](https://user-images.githubusercontent.com/12833922/222963010-8a1c7649-e1ed-4088-8e33-31041b445b0e.png)

**Comment by renanpaiva64** [comment]
I've already accomplished the task. 

Thanks.

**Reply by KieranMaclagan** [reply]
Hey @renanpaiva64, can you please elaborate on what you did to get the SSID? I ran into this as well.

**Reply by plaurent-dev** [reply]
Hi @renanpaiva64, i'm also interested in this !

**Reply by Kristoffeh** [reply]
Don't you just love it when people ask a question and figure out a solution, but don't provide the answer?
Yikes.

**Comment by AlejoPrats** [comment, marked as answer]
Just because I hate to find unanswered questions with this kind of message I will add the answer here for future people looking on how to do this...

remember that you need to have the location enabled

from API 27 (Android 8.1) you need to have just ACCESS_COARSE_LOCATION but from API 29 (Android 10) and on you need both ACCESS_COARSE_LOCATION and ACCESS_FINE_LOCATION

```
        WifiManager wifiManager = (WifiManager)Android.App.Application.Context.GetSystemService(Context.WifiService);
        WifiInfo info = wifiManager.ConnectionInfo;
        String ssid = info.SSID;
```

**Reply by mg-khawaja** [reply]
@renanpaiva64 please mark this as an answer

**Reply by renanpaiva64** [reply]
Tanks
